### PR TITLE
Fixes for times (including properly typed now), room counting, DRY code, etc

### PR DESCRIPTION
**Pull request by 0xdevalias**
This should remove some bugs in how we were using times (eg.`currentTimeInUnixEpoch` being a `const` rather than a function), properly type `utils/time` (instead of it being `.js`), de-dupe a bunch of room entry/counting related code, maybe fix up some deficiencies there, etc. See commits for full log of changes.

**Note:** I haven't done any clickthrough testing with this, so that would definitely be worth doing before merging/releasing.

### CODE REVIEW

File: src/types/Venue.ts
Comment (thecadams) [inline src/types/Venue.ts:24]: just an FYI `AvatarGridRoom` is part of `avatargrid` template which will go soon.

File: src/utils/useLocationUpdateEffect.ts
Comment (thecadams) [inline src/utils/useLocationUpdateEffect.ts:46]: curiosity: did you hold off on this rename to keep the PR from exploding across many files?

File: src/utils/useLocationUpdateEffect.ts
Comment (thecadams) [inline src/utils/useLocationUpdateEffect.ts:97]: You should call `updateLocationData` with just the venue to perform the update.

**Review by thecadams — APPROVED** [review]
Only one comment on the missing attendance data update comment you mentioned, this should be addressed; otherwise looks good at a glance. Approving so as not to block, once you've fixed the one issue in `openRoomWithCounting` feel free to merge.

File: src/types/Venue.ts
Comment (0xdevalias) [reply src/types/Venue.ts:24]: _nods_ all good. This is allowing the types to work for the changes made here in the meantime. Once we 💥  Avatar grid, this can get removed at the same time.

File: src/components/molecules/LiveSchedule/LiveEvent.tsx
Comment (0xdevalias) [reply src/components/molecules/LiveSchedule/LiveEvent.tsx:20]: [OOS] minor changes to existing files, can be cleaned up in a future PR

File: src/components/templates/AvatarGrid/RoomModal.tsx
Comment (0xdevalias) [reply src/components/templates/AvatarGrid/RoomModal.tsx:60]: [OOS] minor changes to existing files, can be cleaned up in a future PR

File: src/utils/time.ts
Comment (0xdevalias) [reply src/utils/time.ts:19]: [OOS] minor changes to existing files, can be cleaned up in a future PR

File: src/utils/useLocationUpdateEffect.ts
Comment (0xdevalias) [reply src/utils/useLocationUpdateEffect.ts:46]: Partially that, and partially because I technically already did it in https://github.com/sparkletown/sparkle/pull/840

https://github.com/sparkletown/sparkle/pull/840/files?file-filters%5B%5D=.ts&hide-deleted-files=true#diff-dc94dd179cbd4f837ed8252c9624d3c3fca4b5e6caefdb499d77616685870dcbR36

File: src/utils/useLocationUpdateEffect.ts
Comment (0xdevalias) [reply src/utils/useLocationUpdateEffect.ts]: resolved in https://github.com/sparkletown/sparkle/pull/897/commits/cb52c8613dec0c7cff6c6c143878245ddbcddfad

File: src/utils/useLocationUpdateEffect.ts
Comment (0xdevalias) [reply src/utils/useLocationUpdateEffect.ts]: resolved in https://github.com/sparkletown/sparkle/pull/897/commits/cb52c8613dec0c7cff6c6c143878245ddbcddfad

File: src/utils/useLocationUpdateEffect.ts
Comment (0xdevalias) [reply src/utils/useLocationUpdateEffect.ts]: resolved in https://github.com/sparkletown/sparkle/pull/897/commits/7362fb9322a1ab624aac307cffc111da7b9b07fe

File: src/components/templates/AvatarGrid/RoomModal.tsx
Comment (0xdevalias) [reply src/components/templates/AvatarGrid/RoomModal.tsx:60]: OOS

File: src/components/molecules/LiveSchedule/LiveEvent.tsx
Comment (0xdevalias) [reply src/components/molecules/LiveSchedule/LiveEvent.tsx:20]: OOS

**Pull request merged**
